hooks/useCollectionCreate.ts

```diff
@@ -1,11 +1,160 @@
+import { useState, useCallback } from "react";
+import { Address } from "viem";
+import { usePrivy } from "@privy-io/react-auth";
+import { useQueryClient } from "@tanstack/react-query";
+import { uploadJson } from "@/lib/arweave/uploadJson";
+import { uploadFilesToArweave } from "@/lib/metadata/uploadFilesToArweave";
 import { toast } from "sonner";
+import { useUserProvider } from "@/providers/UserProvider";
+import { useCollectionsSelection } from "@/hooks/useCollectionsSelection";
+import { useCollectionFormProvider } from "@/providers/CollectionFormProvider";
+import { CHAIN_ID } from "@/lib/consts";
+import { callCreateCollectionApi } from "@/lib/collections/callCreateCollectionApi";
+import type { CollectionsResponse } from "@/types/collections";
 
 export const useCollectionCreate = () => {
-  const handleSubmit = async () => {
-    toast.info("We are cooking this feature right now. Please check back later.");
-  };
+  const [isCreating, setIsCreating] = useState(false);
+  const [uploadProgress, setUploadProgress] = useState(0);
+  const { getAccessToken } = usePrivy();
+  const { artistWallet } = useUserProvider();
+  const { handleValueChange } = useCollectionsSelection();
+  const queryClient = useQueryClient();
+  const { name, imageFile, resetForm, setIsCreateModalOpen } = useCollectionFormProvider();
+
+  const createCollection = useCallback(async (): Promise<{
+    contractAddress: Address;
+    metadataUri: string;
+  } | null> => {
+    if (!name.trim()) {
+      toast.error("Collection name is required");
+      return null;
+    }
+
+    if (!imageFile) {
+      toast.error("Collection image is required");
+      return null;
+    }
+
+    if (!artistWallet) {
+      toast.error("Please connect your artist wallet");
+      return null;
+    }
+
+    setIsCreating(true);
+    setUploadProgress(0);
+
+    try {
+      // Upload image to Arweave
+      const imageUploadResult = await uploadFilesToArweave(
+        null, // previewFile
+        imageFile, // imageFile
+        null, // animationFile
+        "", // existingAnimationUrl
+        setUploadProgress,
+        imageFile.type
+      );
+
+      const imageUri = imageUploadResult.image || imageUploadResult.uploadedImageUri;
+      if (!imageUri) {
+        throw new Error("Failed to upload collection image");
+      }
+
+      // Create collection metadata JSON
+      const metadata = {
+        name,
+        image: imageUri,
+        description: "",
+      };
+
+      // Upload metadata JSON to Arweave
+      setUploadProgress(50);
+      const metadataUri = await uploadJson(metadata);
```

Comment:
Single Responsibility Principle
- actual: metadata creation is handled in the collection creation function
- required: new lib / hook for metadataUri creation
- MIQ: how can **collection** metadata creation use the same hooks / libs as **moment** metadata creation?